src/test/java/net/snowflake/ingest/FipsCompliantIngestIT.java

```diff
@@ -0,0 +1,328 @@
+
+    //create the target
+    IngestExampleHelper.doQuery(conn, "create or replace table "
+        + TABLE
+        + " (row_id int, row_str string, num int, src string)");
+    // Create the pipe for subsequently ingesting files to.
+    IngestExampleHelper.doQuery(conn, "create or replace pipe "
+        + PIPE + " as copy into " + TABLE
+        + " from @" + STAGE
+        + " file_format=(type='csv')");
+
+    String publicKey = IngestExampleHelper.getPublicKeyString(keypair);
+
+    //assume the necessary privileges
+    IngestExampleHelper.doQuery(conn, "use role accountadmin");
+
+    //set the public key
+    IngestExampleHelper.doQuery(conn, "alter user " + USER +
+        " set RSA_PUBLIC_KEY='" + publicKey + "'");
+    files.forEach(file->
+                  { IngestExampleHelper.doQuery(conn, "PUT " + filesLocation
+                      + file + " @" + STAGE +
+                      " AUTO_COMPRESS=FALSE");});
+
+    IngestExampleHelper.doQuery(conn, "use role sysadmin");
+  }
+
+  /**
+   * Generates an RSA keypair for use in this test, with fips provider.
+   *
+   * @return a valid RSA keypair
+   * @throws NoSuchAlgorithmException if we don't have an RSA algo
+   * @throws NoSuchProviderException  if we can't use SHA1PRNG for randomization
+   */
+  public static KeyPair generateKeyPairWithBCFipsProvider()
+      throws NoSuchProviderException, NoSuchAlgorithmException,
+             InvalidAlgorithmParameterException
+  {
+    KeyPairGenerator keyPair = KeyPairGenerator.getInstance(ALGORITHM, "BCFIPS");
+    keyPair.initialize(new RSAKeyGenParameterSpec(2048,
+                                                  RSAKeyGenParameterSpec.F4));
+    return keyPair.generateKeyPair();
+  }
+
+  /**
+   * Ingest a set of files from the staging area through the ingest service
+   * @param files
+   * @return
+   * @throws Exception
+   */
+  private static IngestResponse insertFiles(Set<String> files)
+      throws Exception
+  {
+    return manager.ingestFiles(manager.wrapFilepaths(files), null);
+  }
+
+  /**
+   * Given a set of files, keep querying history until all of files that are
+   * in the set have been seen in the history. Times out after 2 minutes
+   * @param files
+   * @return HistoryResponse containing only the FileEntries corresponding to
+   *          requested input
+   * @throws Exception
+   */
+  private static HistoryResponse waitForFilesHistory(Set<String> files)
+      throws Exception
+  {
+    ExecutorService service = Executors.newSingleThreadExecutor();
+
+    class GetHistory implements
+                     Callable<HistoryResponse>
+    {
+      private Set<String> filesWatchList;
+      GetHistory(Set<String> files)
+      {
+        this.filesWatchList = files;
+      }
+      String beginMark = null;
+
+      public HistoryResponse call()
+          throws Exception
+      {
+        HistoryResponse filesHistory = null;
+        while (true)
+        {
+          Thread.sleep(500);
```

Comment:
Nit - use constants of this an other numbers, paths, ect. you directly use in the code.